primitives/state-machine/src/lib.rs

```diff
@@ -349,7 +347,7 @@ impl<'a, B, H, N, Exec> StateMachine<'a, B, H, N, Exec> where
 		let (result, was_native) = self.execute_aux(true, native_call.take());
 
 		if was_native {
-			self.overlay.prospective = orig_prospective.clone();
+			self.overlay.discard_prospective();
```

Comment:
This is wrong, we need to restore the other prospective changes.

Reply:
I reverted the change.